doc/development/adding_templates.rst

```diff
@@ -0,0 +1,91 @@
+     it implements.
+
+   * A template is "registered" by using the :mod:`template <pennylane.template>` decorator ``@template``.
+     This allows us to record the queue of operations of a template, which is very useful for testing:
+
+     .. code-block:: python
+
+        with qml.utils.OperationRecorder() as rec:
+            MyNewTemplate(...)
+
+        list_of_gates = rec.queue
+
+   * Your template should have a positional argument called ``wires``, which takes an integer (single wire) or a list
+     of wire indices that the template acts on.
+
+   * Consider using the :func:`broadcasting <pennylane.templates.broadcasting>` function to make your code more concise.
+
+   * Write an extensive docstring that explains how to use the template. Include a sketch of the template (add the
+     file to the ``doc/_static/templates/<templ_type>/`` directory). You can also display a small usage example
+     at the beginning of the docstring.
+
+     At the end of the docstring, add a section starting with the ``.. UsageDetails::`` directive,
+     where you demonstrate with code examples how to use the templates with different
+     settings, for example varying the number of wires, explaining keyword arguments and special cases.
+
+   * Check the inputs to the template. You can use the functions provided in :mod:`utils <pennylane.templates.utils>`.
+     Don't forget that arguments may be passed by the user to the qnode as positional or keyword arguments, and
+     by using different interfaces (i.e., a input could be a ``numpy.ndarray`` or a list of
+     :class:`Variable <pennylane.variable.Variable>`, depending on how the user uses the template).
+
+2. **Import the new template** in ``templates/<templ_type>/__init__.py`` by adding the new line
+
+   .. code-block:: python
+
+        from .mynewtemplate import MyNewTemplate
+
+3. **Add your template to the documentation** by adding a ``customgalleryitem`` to the correct layer type section in
+   ``doc/introduction/templates.rst``:
+
+   .. code-block::
+
+     .. customgalleryitem::
+        :link: ../code/api/pennylane.templates.<templ_type>.MyNewTemplate.html
+        :description: MyNewTemplate
+        :figure: ../_static/templates/<templ_type>/my_new_template.png
+
+   .. note::
+
+      This loads the image of the template added to ``doc/_static/templates`` in Step 1. Make sure that
+      this image has the same dimensions and style than other template icons.
+
+4. **Add tests** for your new template to the test suite.
+
+   * Integration tests which check that your template can be called inside a quantum node, and that PennyLane can
+     compute gradients with respect to differentiable parameters, are added to ``tests/test_templates.py``.
+     Simply add your template to the fixtures (variables indicated by capital letters) to automatically
+     run existing tests on your new template.
```

Comment:
This may need more explanation 😆

Reply:
I was thinking to add the explanation to the test file, so it is easier to maintain?

Reply:
[I thought I replied, but seems it didn't go through...]

For easy maintenance I'd love to put the explanation into the test file itself!